cranelift/codegen/src/ir/globalvalue.rs

```diff
@@ -76,6 +76,13 @@ pub enum GlobalValueData {
         /// Does this symbol refer to a thread local storage value?
         tls: bool,
     },
+
+    /// Value is a multiple of how many instances of `vector_type` will fit in
+    /// a target vector register.
```

Comment:
This is slightly unclear to me: does this mean that it takes the size of the target's register width and divides by `vector_type`? i.e. that it has a concrete value determined by the compiler configuration / target microarchitecture? Or is it passed in / determined in some other way?

Or I guess my confusion comes from: I would expect `DynScale` as a name to imply that this is some generic sort of scale parameter, while this seems to say that it is a specific scale value determined by some algorithm ("what will fit in ..."). So maybe a better name would help: `DynScaleTargetReg`? And in the future we may have some other `DynScale...` options based on loads of parameters from vmctx, or a special vector-length register, or a `DynScaleConstant`, or ...

Reply:
> does this mean that it takes the size of the target's register width and divides by vector_type?

Yes. So for a given dynamic type, the backend will scale the fixed part of the type by `DynScale`. The name is likely misleading.

I feel we shouldn't need to prescribe different global values for constant scale or truly dynamic ones via a register, as it's not immediately obvious, to me, why this would be helpful at the IR level. I feel we should be able to encapsulate the semantics we want with a single construct and allow a backend to lower it how it wishes.... but...

It would almost certainly make it far easier to implement the machinst ABI and, as that is a particular point of pain, it would be good to make that as simple as possible :)

For this first implementation we could rename to `DynScaleConstant` and it would much better describe the current limitations.

Reply:
Yes, I think `DynScaleTargetWidth` or something might be best here. As to why different global values for the different cases -- if I'm understanding the question right, the answer is that the details are really ABI-visible (e.g., is there a hidden register that controls how much data we read at once that must be set up? is it part of a vmcontext in a runtime?) so they should be surfaced in the IR. And having it clearly here also lets us define CLIF semantics for it, which is useful for interpreters and for formal verification.

Anyway with the above rename I think this is settled for now!